From: =?utf-8?q?Mantas_Mikul=C4=97nas?= <grawity@gmail.com>
Date: Wed, 13 Nov 2013 13:36:17 +0200
Subject: activate: mention -E in the help text

---
 src/activate/activate.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/activate/activate.c b/src/activate/activate.c
index 83d25b13..fc581b92 100644
--- a/src/activate/activate.c
+++ b/src/activate/activate.c
@@ -316,6 +316,8 @@ static int help(void) {
                "  -l --listen=ADDR     Listen for raw connections at ADDR\n"
                "  -a --accept          Spawn separate child for each connection\n"
                "  -h --help            Show this help and exit\n"
+               "  -E --environment=NAME[=VALUE]\n"
+               "                       Pass an environment variable to children\n"
                "  --version            Print version string and exit\n"
                "\n"
                "Note: file descriptors from sd_listen_fds() will be passed through.\n"
