From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Mon, 8 Sep 2014 13:04:01 +0200
Subject: Add support for HP ProLiant m400 Server Cartridge soft powerdown

LP: #1354306
---
 src/login/70-power-switch.rules | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/login/70-power-switch.rules b/src/login/70-power-switch.rules
index 36fb827..a11c9f4 100644
--- a/src/login/70-power-switch.rules
+++ b/src/login/70-power-switch.rules
@@ -9,5 +9,7 @@ ACTION=="remove", GOTO="power_switch_end"
 
 SUBSYSTEM=="input", KERNEL=="event*", SUBSYSTEMS=="acpi", TAG+="power-switch"
 SUBSYSTEM=="input", KERNEL=="event*", KERNELS=="thinkpad_acpi", TAG+="power-switch"
+SUBSYSTEM=="input", KERNEL=="event*", SUBSYSTEMS=="platform", KERNELS=="gpio_keys.6", PROGRAM="/bin/cat /proc/device-tree/model", RESULT=="HP ProLiant m400 Server Cartridge", TAG+="power-switch"
+SUBSYSTEM=="input", KERNEL=="event*", SUBSYSTEMS=="platform", KERNELS=="gpio_keys.12", ATTRS{keys}=="116", PROGRAM="/bin/cat /proc/device-tree/model", RESULT=="HP ProLiant m800 Server Cartridge", TAG+="power-switch"
 
 LABEL="power_switch_end"
