From: Michael Stapelberg <michael@stapelberg.de>
Date: Thu, 8 Aug 2013 21:34:12 +0200
Subject: change the log_info() line in systemctl to reflect reality

---
 src/systemctl/systemctl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/systemctl/systemctl.c b/src/systemctl/systemctl.c
index a008517..d70657c 100644
--- a/src/systemctl/systemctl.c
+++ b/src/systemctl/systemctl.c
@@ -4085,7 +4085,7 @@ static int enable_sysv_units(char **args) {
                 /* Mark this entry, so that we don't try enabling it as native unit */
                 args[f] = (char*) "";
 
-                log_info("%s is not a native service, redirecting to /usr/sbin/update-rc.d.", name);
+                log_info("Synchronizing state for %s with sysvinit using update-rc.d...", name);
 
                 argv[c++] = path_get_file_name(p);
                 argv[c++] = streq(verb, "enable") ? "defaults" : "remove";
