From: =?utf-8?q?Zbigniew_J=C4=99drzejewski-Szmek?= <zbyszek@in.waw.pl>
Date: Thu, 5 Dec 2013 16:06:04 -0500
Subject: core/device: ignore SYSTEMD_WANTS= in user mode

---
 src/core/device.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/core/device.c b/src/core/device.c
index 610fe88e..e02c207c 100644
--- a/src/core/device.c
+++ b/src/core/device.c
@@ -275,8 +275,9 @@ static int device_update_unit(Manager *m, struct udev_device *dev, const char *p
                         }
                 }
 
-                wants = udev_device_get_property_value(dev, "SYSTEMD_WANTS");
-                if (wants) {
+                if (u->manager->running_as == SYSTEMD_SYSTEM &&
+                    (wants = udev_device_get_property_value(dev, "SYSTEMD_WANTS"))) {
+
                         char *state, *w;
                         size_t l;
 
