From: David Strauss <david@davidstrauss.net>
Date: Thu, 17 Oct 2013 13:19:29 -0700
Subject: Fix kmod error message to have correct version requirement

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 5bc31c5..9904e25 100644
--- a/configure.ac
+++ b/configure.ac
@@ -271,7 +271,7 @@ if test "x$enable_kmod" != "xno"; then
         if test "x$have_kmod" = "xyes"; then
                 PKG_CHECK_MODULES(KMOD, [ libkmod >= 15 ],
                         [AC_DEFINE(HAVE_KMOD, 1, [Define if kmod is available])],
-                        AC_MSG_ERROR([*** kmod version >= 14 not found]))
+                        AC_MSG_ERROR([*** kmod version >= 15 not found]))
         fi
         if test "x$have_kmod" = xno -a "x$enable_kmod" = xyes; then
                 AC_MSG_ERROR([*** kmod support requested, but libraries not found])
