From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Wed, 12 Aug 2015 15:03:52 +0200
Subject: Hack to unbreak "systemctl link"

Bug: https://github.com/systemd/systemd/issues/879
Bug-Ubuntu: https://launchpad.net/bugs/1480310
---
 src/core/dbus-manager.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/core/dbus-manager.c b/src/core/dbus-manager.c
index 5722e3c..ce036de 100644
--- a/src/core/dbus-manager.c
+++ b/src/core/dbus-manager.c
@@ -1650,9 +1650,11 @@ static int method_enable_unit_files_generic(
         if (r < 0)
                 return r;
 
-        r = mac_selinux_unit_access_check_strv(l, message, m, verb, error);
-        if (r < 0)
-                return r;
+        if (call != unit_file_link) {
+                r = mac_selinux_unit_access_check_strv(l, message, m, verb, error);
+                if (r < 0)
+                        return r;
+        }
 
         r = bus_verify_manage_unit_files_async(m, message, error);
         if (r < 0)
