From: Lennart Poettering <lennart@poettering.net>
Date: Wed, 20 May 2015 14:37:21 +0200
Subject: journald: handle more gracefully when bind() fails on audit sockets

---
 src/journal/journald-audit.c | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/src/journal/journald-audit.c b/src/journal/journald-audit.c
index 46eb82f..0e739a2 100644
--- a/src/journal/journald-audit.c
+++ b/src/journal/journald-audit.c
@@ -532,9 +532,14 @@ int server_open_audit(Server *s) {
                         return 0;
                 }
 
-                r = bind(s->audit_fd, &sa.sa, sizeof(sa.nl));
-                if (r < 0)
-                        return log_error_errno(errno, "Failed to join audit multicast group: %m");
+                if (bind(s->audit_fd, &sa.sa, sizeof(sa.nl)) < 0) {
+                        log_warning_errno(errno,
+                                          "Failed to join audit multicast group. "
+                                          "The kernel is probably too old or multicast reading is not supported. "
+                                          "Ignoring: %m");
+                        s->audit_fd = safe_close(s->audit_fd);
+                        return 0;
+                }
         } else
                 fd_nonblock(s->audit_fd, 1);
 
