From: Michal Sekletar <msekleta@redhat.com>
Date: Mon, 10 Mar 2014 18:26:34 +0100
Subject: logind: save user state after stopping the session

---
 src/login/logind-session.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/login/logind-session.c b/src/login/logind-session.c
index 3ab41e63..2a2a6704 100644
--- a/src/login/logind-session.c
+++ b/src/login/logind-session.c
@@ -685,6 +685,8 @@ int session_stop(Session *s) {
         s->stopping = true;
 
         session_save(s);
+        user_save(s);
+
         return r;
 }
 
