From: Michal Sekletar <msekleta@redhat.com>
Date: Tue, 11 Mar 2014 13:26:38 +0100
Subject: logind: initialize timer_fd

---
 src/login/logind-session.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/login/logind-session.c b/src/login/logind-session.c
index 2a2a670..d7c3187 100644
--- a/src/login/logind-session.c
+++ b/src/login/logind-session.c
@@ -89,6 +89,7 @@ Session* session_new(Manager *m, const char *id) {
 
         s->manager = m;
         s->fifo_fd = -1;
+        s->timer_fd = -1;
 
         return s;
 }
