From: Michal Sekletar <msekleta@redhat.com>
Date: Tue, 11 Mar 2014 20:42:42 +0100
Subject: logind: pass pointer to User object to user_save()

---
 src/login/logind-session.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/login/logind-session.c b/src/login/logind-session.c
index d7c3187..b0e4bf6 100644
--- a/src/login/logind-session.c
+++ b/src/login/logind-session.c
@@ -686,7 +686,7 @@ int session_stop(Session *s) {
         s->stopping = true;
 
         session_save(s);
-        user_save(s);
+        user_save(s->user);
 
         return r;
 }
