From: Michal Sekletar <msekleta@redhat.com>
Date: Thu, 13 Mar 2014 11:18:27 +0100
Subject: logind: add forgotten return statement

---
 src/login/logind-session.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/login/logind-session.c b/src/login/logind-session.c
index 730f11de..ece222af 100644
--- a/src/login/logind-session.c
+++ b/src/login/logind-session.c
@@ -782,6 +782,8 @@ void session_release(Session *s) {
                 goto out;
         }
 
+        return;
+
 out:
         if (s->timer_fd >= 0) {
                 close_nointr(s->timer_fd);
