Description: Add power button support for HP Moonshot cartridges
Author: Manoj Iyer <manoj.iyer@canonical.com>
Author: dann frazier <dann.frazier@canonical.com>
Bug-Ubuntu: http://launchpad.net/bugs/1354306,
Bug-Ubuntu: http://launchpad.net/bugs/1347776

Index: systemd-204/src/login/70-power-switch.rules
===================================================================
--- systemd-204.orig/src/login/70-power-switch.rules
+++ systemd-204/src/login/70-power-switch.rules
@@ -9,5 +9,7 @@ ACTION=="remove", GOTO="power_switch_end
 
 SUBSYSTEM=="input", KERNEL=="event*", SUBSYSTEMS=="acpi", TAG+="power-switch"
 SUBSYSTEM=="input", KERNEL=="event*", KERNELS=="thinkpad_acpi", TAG+="power-switch"
+SUBSYSTEM=="input", KERNEL=="event*", SUBSYSTEMS=="platform", KERNELS=="gpio_keys.6", PROGRAM="/bin/cat /proc/device-tree/model", RESULT=="HP ProLiant m400 Server Cartridge", TAG+="power-switch"
+SUBSYSTEM=="input", KERNEL=="event*", SUBSYSTEMS=="platform", KERNELS=="gpio_keys.12", ATTRS{keys}=="116", PROGRAM="/bin/cat /proc/device-tree/model", RESULT=="HP ProLiant m800 Server Cartridge", TAG+="power-switch"
 
 LABEL="power_switch_end"
