From: Michael Stapelberg <michael@stapelberg.de>
Date: Sat, 10 Aug 2013 11:12:52 +0200
Subject: systemctl: call update-rc.d enable|disable, not defaults|remove

---
 src/systemctl/systemctl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/systemctl/systemctl.c b/src/systemctl/systemctl.c
index e3c2b19..42c1d13 100644
--- a/src/systemctl/systemctl.c
+++ b/src/systemctl/systemctl.c
@@ -4303,7 +4303,7 @@ static int enable_sysv_units(const char *verb, char **args) {
                 log_info("Synchronizing state for %s with sysvinit using update-rc.d...", name);
 
                 argv[c++] = path_get_file_name(p);
-                argv[c++] = streq(verb, "enable") ? "defaults" : "remove";
+                argv[c++] = verb;
                 argv[c] = NULL;
 
                 l = strv_join((char**)argv, " ");
