From: Harald Hoyer <harald@redhat.com>
Date: Wed, 3 Jun 2015 16:36:20 +0200
Subject: test-unit-file.c: fixup the test for commit 3b51f8ddd5

---
 src/test/test-unit-file.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/test/test-unit-file.c b/src/test/test-unit-file.c
index 31b12d5..a8025c8 100644
--- a/src/test/test-unit-file.c
+++ b/src/test/test-unit-file.c
@@ -232,7 +232,7 @@ static void test_config_parse_exec(void) {
                               &c, NULL);
         assert_se(r >= 0);
         c1 = c1->command_next;
-        check_execcommand(c1, "/bin/find", NULL, "\\073", NULL, false);
+        check_execcommand(c1, "/bin/find", NULL, ";", NULL, false);
 
         log_info("/* spaces in the filename */");
         r = config_parse_exec(NULL, "fake", 5, "section", 1,
