From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Wed, 21 Jan 2015 15:37:40 +0100
Subject: util: Add hidden suffixes for ucf

Bug-Debian: https://bugs.debian.org/775903
---
 src/basic/path-util.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/basic/path-util.c b/src/basic/path-util.c
index ec90c43..ce38acf 100644
--- a/src/basic/path-util.c
+++ b/src/basic/path-util.c
@@ -778,6 +778,9 @@ bool hidden_file_allow_backup(const char *filename) {
                 endswith(filename, ".dpkg-bak") ||
                 endswith(filename, ".dpkg-backup") ||
                 endswith(filename, ".dpkg-remove") ||
+                endswith(filename, ".ucf-new") ||
+                endswith(filename, ".ucf-old") ||
+                endswith(filename, ".ucf-dist") ||
                 endswith(filename, ".swp");
 }
 
