From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Wed, 21 Jan 2015 15:37:40 +0100
Subject: util: Add hidden suffixes for ucf

Bug-Debian: https://bugs.debian.org/775903
---
 src/basic/util.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/basic/util.c b/src/basic/util.c
index 2855993..a055ea2 100644
--- a/src/basic/util.c
+++ b/src/basic/util.c
@@ -1927,6 +1927,9 @@ _pure_ static bool hidden_file_allow_backup(const char *filename) {
                 endswith(filename, ".dpkg-bak") ||
                 endswith(filename, ".dpkg-backup") ||
                 endswith(filename, ".dpkg-remove") ||
+                endswith(filename, ".ucf-new") ||
+                endswith(filename, ".ucf-old") ||
+                endswith(filename, ".ucf-dist") ||
                 endswith(filename, ".swp");
 }
 
