From: Michal Sekletar <msekleta@redhat.com>
Date: Mon, 10 Feb 2014 15:41:47 +0100
Subject: utils: silence the compiler warning

---
 src/shared/util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shared/util.c b/src/shared/util.c
index 52451f9..7500779 100644
--- a/src/shared/util.c
+++ b/src/shared/util.c
@@ -427,7 +427,7 @@ char *split_quoted(const char *c, size_t *l, char **state) {
                 *state = (char*) e;
         }
 
-        return current;
+        return (char *) current;
 }
 
 int get_parent_of_pid(pid_t pid, pid_t *_ppid) {
